src/engines/dragonage/object.cpp

```diff
@@ -47,6 +50,9 @@ using ::Aurora::GFF4List;
 using namespace ::Aurora::GFF4FieldNamesEnum;
 
 Object::Object(ObjectType type) : _type(type), _static(true), _usable(false) {
+	_id = Common::generateIDNumber();
```

Comment:
For areas, the _id gets overwritten in Area::loadEnvironment(), with a value from the GFF. You'll need to add a commit adding a, for example _envID value to Area and change the assignment in loadEnvironment to use write into that value instead.

Same applies to Dragon Age II as well.